### ISSUE TITLE
False positive in `implicit_assignment_linter()` for code with pipe

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1827]
``` r
library(lintr)

lint(
  text = "mutate(data, a := b)",
  linters = implicit_assignment_linter()
)


lint(
  text = "data %>% mutate(a := b)",
  linters = implicit_assignment_linter()
)
#> <text>:1:19: warning: [implicit_assignment_linter] Avoid implicit assignments in function calls. For example, instead of `if (x <- 1L) { ... }`, write `x <- 1L; if (x) { ... }`.
#> data %>% mutate(a := b)
#>                   ^~
```

<sup>Created on 2022-12-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
